### PR TITLE
Migrate to Core

### PR DESCRIPTION
**Pull request by aspeddro**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ This is the official documentation platform for the [ReScript](https://rescript-
 
 ## System Requirements
 
-- `node@18` or higher
+- `node@20` or higher
```
Comment (aspeddro) [inline README.md:19]: Core has bindings to `Array.toSorted`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSorted require node v20

File: src/common/Util.res [hunk 2]
```diff
@@ -53,11 +53,6 @@ module String = {
     }")
 }
 
-module Json = {
-  @val @scope("JSON")
-  external prettyStringify: (Js.Json.t, @as(json`null`) _, @as(4) _) => string = "stringify"
-}
-
```
Comment (aspeddro) [inline src/common/Util.res:60]: We can use `JSON.stringify(json, ~space=2)`

File: src/components/SearchBox.res [hunk 3]
```diff
@@ -18,19 +18,20 @@ let make = (
...
-  let focusInput = () =>
-    textInput.current->Js.Nullable.toOption->Belt.Option.forEach(el => el->focus)
+  let focusInput = () => textInput.current->Nullable.toOption->Option.forEach(el => el->focus)
 
   let onAreaFocus = evt => {
     let el = ReactEvent.Focus.target(evt)
-    let isDiv = Js.Null_undefined.isNullable(el["type"])
+    // TODO(aspeddro)
+    // let isDiv = Js.Null_undefined.isNullable(el["type"])
+    let isDiv = Nullable.toOption(el["type"])->Option.isSome
```
[the remark this answers is not in the document]
Comment (fhammerschmidt) [reply src/components/SearchBox.res:34]: not really, as it is not optimized by the compiler in the null case
I am not really sure why this did not make it into Core, @zth ?

[the remark this answers is not in the document]
Comment (zth) [reply src/components/SearchBox.res:34]: Which one?

[the remark this answers is not in the document]
Comment (fhammerschmidt) [reply src/components/SearchBox.res:34]: `Js.Null_undefined.isNullable`

[the remark this answers is not in the document]
Comment (fhammerschmidt) [reply src/components/SearchBox.res:34]: it's just
`external isNullable: t<'a> => bool = "#is_nullable"`

[the remark this answers is not in the document]
Comment (zth) [reply src/components/SearchBox.res:34]: Ah, probably just a miss. Care to PR it?

**Review by fhammerschmidt — CHANGES_REQUESTED** [review]

File: pages/docs/react/latest/forwarding-refs.mdx [hunk 4]
```diff
@@ -56,12 +56,12 @@ module FancyInput = {
...
 @react.component
 let make = () => {
-  let input = React.useRef(Js.Nullable.null)
+  let input = React.useRef(Nullable.null)
 
   let focusInput = () =>
     input.current
-    ->Js.Nullable.toOption
-    ->Belt.Option.forEach(input => input->focus)
+    ->Nullable.toOption
```
[the remark this answers is not in the document]
Comment (aspeddro) [reply pages/docs/react/latest/forwarding-refs.mdx]: Done https://github.com/rescript-association/rescript-lang.org/pull/876/commits/6d659e5007171c20dae3759fdcaf27b0b7004b67

File: pages/docs/react/latest/hooks-ref.mdx [hunk 5]
```diff
@@ -56,12 +56,12 @@ More infos on direct DOM manipulation can be found in the [Refs and the DOM](./r
...
 @react.component
 let make = () => {
-  let inputEl = React.useRef(Js.Nullable.null)
+  let inputEl = React.useRef(Nullable.null)
 
   let onClick = _ => {
     inputEl.current
-    ->Js.Nullable.toOption
-    ->Belt.Option.forEach(input => input->focus)
+    ->Nullable.toOption
```
[the remark this answers is not in the document]
Comment (aspeddro) [reply pages/docs/react/latest/hooks-ref.mdx]: Done https://github.com/rescript-association/rescript-lang.org/pull/876/commits/6d659e5007171c20dae3759fdcaf27b0b7004b67

File: pages/docs/react/latest/hooks-ref.mdx [hunk 6]
```diff
@@ -104,15 +104,15 @@ Reusing the example from our [Refs and the DOM](./refs-and-the-dom#callback-refs
...
+  let textInput = React.useRef(Nullable.null)
   let setTextInputRef = element => {
     textInput.current = element;
   }
 
   let focusTextInput = _ => {
     textInput.current
-    ->Js.Nullable.toOption
-    ->Belt.Option.forEach(input => input->focus)
+    ->Nullable.toOption
```
[the remark this answers is not in the document]
Comment (aspeddro) [reply pages/docs/react/latest/hooks-ref.mdx]: Done https://github.com/rescript-association/rescript-lang.org/pull/876/commits/6d659e5007171c20dae3759fdcaf27b0b7004b67

File: pages/docs/react/latest/refs-and-the-dom.mdx [hunk 7]
```diff
@@ -83,10 +83,10 @@ This code uses a `React.ref` to store a reference to an `input` DOM node to put
 
 @react.component
 let make = () => {
-  let textInput = React.useRef(Js.Nullable.null)
+  let textInput = React.useRef(Nullable.null)
 
   let focusInput = () =>
-    switch textInput.current->Js.Nullable.toOption {
+    switch textInput.current->Nullable.toOption {
```
[the remark this answers is not in the document]
Comment (aspeddro) [reply pages/docs/react/latest/refs-and-the-dom.mdx]: Done https://github.com/rescript-association/rescript-lang.org/pull/876/commits/6d659e5007171c20dae3759fdcaf27b0b7004b67

File: pages/docs/react/latest/refs-and-the-dom.mdx [hunk 8]
```diff
@@ -193,15 +193,15 @@ The example below implements a common pattern: using the ref callback to store a
...
+  let textInput = React.useRef(Nullable.null)
   let setTextInputRef = element => {
     textInput.current = element;
   }
 
   let focusTextInput = _ => {
     textInput.current
-    ->Js.Nullable.toOption
-    ->Belt.Option.forEach(input => input->focus)
+    ->Nullable.toOption
```
[the remark this answers is not in the document]
Comment (aspeddro) [reply pages/docs/react/latest/refs-and-the-dom.mdx]: Done https://github.com/rescript-association/rescript-lang.org/pull/876/commits/6d659e5007171c20dae3759fdcaf27b0b7004b67

File: src/ApiDocs.res [hunk 9]
```diff
@@ -216,16 +215,16 @@ module SidebarTree = {
...
             {node.name->React.string}
           </Next.Link>
-          {moduleRoute->Js.Array2.length === 1 ? subMenu : React.null}
+          {moduleRoute->Array.length === 1 ? subMenu : React.null}
         </div>
         <div className="hl-overline text-gray-80 mt-5 mb-2"> {"submodules"->React.string} </div>
         {node.children
-        ->Js.Array2.sortInPlaceWith((v1, v2) => v1.name > v2.name ? 1 : -1)
-        ->Js.Array2.map(renderNode)
+        ->Belt.SortArray.stableSortBy((v1, v2) => v1.name > v2.name ? 1 : -1)
```
[the remark this answers is not in the document]
Comment (aspeddro) [reply src/ApiDocs.res]: Done https://github.com/rescript-association/rescript-lang.org/pull/876/commits/6d659e5007171c20dae3759fdcaf27b0b7004b67

File: src/Packages.res [hunk 10]
```diff
@@ -102,7 +100,7 @@ module Resource = {
 
     fuser
     ->Fuse.search(pattern)
-    ->Js.Array2.sortInPlaceWith((a, b) => a["item"].searchScore > b["item"].searchScore ? -1 : 1)
+    ->Belt.SortArray.stableSortBy((a, b) => a["item"].searchScore > b["item"].searchScore ? -1 : 1)
```
[the remark this answers is not in the document]
Comment (aspeddro) [reply src/Packages.res]: Done https://github.com/rescript-association/rescript-lang.org/pull/876/commits/6d659e5007171c20dae3759fdcaf27b0b7004b67

File: src/components/SearchBox.res (hunk 3 above)
[the remark this answers is not in the document]
Comment (aspeddro) [reply src/components/SearchBox.res]: Yes. I Opened a PR on Core. I'll leave `Js.Null Undefined.is Nullable` until a new version of Core is released

**Comment by aspeddro** [comment]
More two commits here:
- https://github.com/rescript-association/rescript-lang.org/pull/876/commits/8bb5c90464fb2c6bf725169f15d05b7e68610c0a
- https://github.com/rescript-association/rescript-lang.org/pull/876/commits/901c244a8557f206e116f45e2244fd7fac9c4c0a

**Review by fhammerschmidt — APPROVED** [review]

**Pull request merged**